internal/controller/api/v1beta1/agentserviceconfig_types.go

```diff
@@ -40,6 +40,12 @@ type AgentServiceConfigSpec struct {
 	// With respect to the resource requests, a minimum of 10Gi is recommended.
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Storage for database"
 	DatabaseStorage corev1.PersistentVolumeClaimSpec `json:"databaseStorage"`
+	// MirrorRegistryConfigMapName defines the name of the configmap that contains mirror registry configuration
+	// In case no configuration is need, this field will be empty. ConfigMap must contain to entries:
+	// ca-bundle.crt - hold the contents of mirror registry certificate/s
+	// registries.conf - holds the content of registries.conf file configured with mirror registries
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Mirror Registry and Certificate ConfigMap Name"
+	MirrorRegistryConfigMapName string `json:"mirrorRegistryConfigmapName,omitempty"`
```

Comment:
Do we want to use a `LocalObjectReference` here instead of a `string` ? 

I have seen a mix of usage between `Refs` and `Names` and it'd be nice to be more consistent. `ClusterDeployment` uses references for cases like this. Any opinions about it?

Here's an example: https://github.com/openshift/hive/blob/d5b066d39c7d00acb05769c4935c0f8ea3554210/apis/hive/v1/clusterdeployment_types.go#L89

Reply:
it is a matter of convention, if we have one. i personally think that Name string is more clear for the user, but i can change it. @djzager @mhrivnak what do you think?

Reply:
If we are using `string` I think `MirrorRegistryConfigMapName` makes sense. If we use `LocalObjectReference` then I believe `MirrorRegistryConfigMapRef` is better. Since I've seen both `Name` and `Ref`, I don't believe that there is an established convention so as long as we are internally consistent I'm happy.

Reply:
I'm not certain that `LocalObjectReference` type buys us very much in this scenario. Is there something I'm missing @flaper87 ? Assuming that using `LocalObjectReference` doesn't add value, I say `string` is fine for us.

Reply:
> I'm not certain that LocalObjectReference type buys us very much in this scenario. Is there something I'm missing @flaper87 ? Assuming that using LocalObjectReference doesn't add value, I say string is fine for us.

The only thing I'm after is consistency. Here we are using strings, but then in other places we use `LocalObjectReference`. https://github.com/openshift/assisted-service/blob/5ddcfc1625de87be709c08f42bc83c73753962ca/internal/controller/api/v1beta1/infraenv_types.go#L63

To be honest, I think using a `LocalObjectReference` is clearer, it's more explicit, and it sets all the expectations and requirements we need. 

> If we are using string I think MirrorRegistryConfigMapName makes sense. If we use LocalObjectReference then I believe MirrorRegistryConfigMapRef is better.

:+1: 

I'm happy to let this one go if others think it's not worth it but it seems to me like we are letting all these cases go in and missing the chance of being consistent. This, for the user, is way worse than the extra characters needed to write down the reference.

Reply:
`LocalObjectReference` is probably the better option and matches the core API. Matching what hive does is also valuable. If we have other fields that are local `*Name`, I'd support changing them to `LocalObjectReference` for consistency.

Reply:
To elaborate, the "better" part is AIUI the fact that context and usage is implicit in the type. We don't have to document for each field "This is the Name of a kube resource in the same namespace as this resource." It's implicit.

Reply:
Wasn't aware of other examples in this repo :fearful:. Since you put it this way this is my vote:

```suggestion
	MirrorRegistryConfigMapRef *corev1.LocalObjectReference `json:"mirrorRegistryConfigmapRef,omitempty"`
```

Reply:
fixed